Gravity/Gravity.NUnit.Integration/RSAPI_IntegrationTest.cs

```diff
@@ -0,0 +1,395 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using kCura.Relativity.Client;
+using NUnit.Framework;
+using Relativity.API;
+using Relativity.Test.Helpers;
+using Relativity.Test.Helpers.ServiceFactory.Extentions;
+using Relativity.Test.Helpers.SharedTestHelpers;
+using Relativity.Test.Helpers.WorkspaceHelpers;
+using Gravity.Test;
+using System.Configuration;
+using System.Diagnostics.Eventing.Reader;
+using Gravity.Base;
+using Gravity.Test.TestClasses;
+using kCura.Relativity.Client.DTOs;
+using System.Reflection;
+using Gravity.Test.Helpers;
+
+namespace Gravity.NUnit.Integration
+{
+    [TestFixture]
+    public class RSAPI_IntegrationTest
+    {
+        #region Variables
+
+        //this is set to true and should skip the creation of the database, installation of the testing application, and the deletion of the workspace.
+        //must point _workspaceId = valid workspace with application already installed in the Setup
+        private bool _debug = Convert.ToBoolean(ConfigurationManager.AppSettings["Debug"]);
+        private int _debugWorkspaceId = Convert.ToInt32(ConfigurationManager.AppSettings["DebugWorkspaceId"]);
+
+        private IRSAPIClient _client;
+        private readonly string _workspaceName = $"GravityTest_{Guid.NewGuid()}";
+        private int _workspaceId;
+        private IServicesMgr _servicesManager;
+        private IDBContext _eddsDbContext;
+        private IDBContext _dbContext;
+        //public string FilepathApplication = TestHelpers.Constants.Agent.DEFAULT_RAP_FILE_LOCATION + TestHelpers.Constants.Application.General.APPLICATION_NAME;
+        public string _applicaitonFilePath = ConfigurationManager.AppSettings["TestApplicationLocation"];
+        public string _applicationName = ConfigurationManager.AppSettings["TestApplicationName"];
+        private Test.Helpers.TestObjectHelper _testObjectHelper;
+        #endregion
+
+        #region Setup
+        [TestFixtureSetUp]
+        public void Execute_TestFixtureSetup()
```

Comment:
This method is too large. Break down into smaller helper methods that could be reused later.